documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -225,6 +225,26 @@
               description: 'Prometheus %(prometheusName)s remote write is {{ printf "%%.1f" $value }}s behind for queue {{$labels.queue}}.' % $._config,
             },
           },
+          {
+            alert: 'PrometheusRemoteWriteDesiredShards',
+            expr: |||
+              # Without max_over_time, failed scrapes could create false negatives, see
+              # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
+              (
+                max_over_time(prometheus_remote_storage_shards_desired{%(prometheusSelector)s}[5m])
+              > on(job, instance) group_right
+                max_over_time(prometheus_remote_storage_shards_max{%(prometheusSelector)s}[5m])
+              )
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              summary: 'Prometheus remote write desired shards calculation wants to run more than configured max shards.',
+              description: 'Prometheus %(prometheusName)s remote write desired shards calculation wants to run {{ printf $value }} shards, which is more than the max of {{ printf `prometheus_remote_storage_shards_max{%(prometheusSelector)s}` | query | first | value }}.' % $._config,
```

Comment:
The query will just pick the first result, not necessarily the one for the Prometheus server affected. I believe it has to be written in the following way:

```yaml
              description: 'Prometheus %(prometheusName)s remote write desired shards calculation wants to run {{ printf $value }} shards, which is more than the max of {{ printf `prometheus_remote_storage_shards_max{instance="%%s",%(prometheusSelector)s}` $labels.instance | query | first | value }}.' % $._config,
```

Reply:
Hmmm okay, is that not what `prometheusSelectors` is supposed to be doing? I assumed those would contain the label selectors for the instance that triggered the alert.

Reply:
`prometheusSelector` selects _all_ Prometheus servers to monitor.

Reply:
In this case, it would intersect with the alerts firing, i.e. if only one Prometheus server fires this alerts, all is good, but as soon as more than one of the monitored Prometheus servers are affected by this alert, you'll get multiple results in the query.

Reply:
:+1: got it, thank you!